### PR TITLE
add copy to clipboard behaviour

### PR DESCRIPTION
**Pull request by earnubs**
spec stipulates that a copy-to-clipboard icon exist below cli instructions, part of #655

### CODE REVIEW

**Comment by magdalena-mirowicz** [comment]
please pause this work for now

**Comment by magdalena-mirowicz** [comment]
@earnubs please ignore me, it's ok to work on

File: src/common/components/share/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { Component, PropTypes } from 'react';
+import Clipboard from 'clipboard';
+
+import styles from './styles.css';
+
+const COPY_TO_CLIPBOARD_CLASS = styles.clipboard;
+
+export class CopyToClipboard extends Component {
+  componentDidMount() {
+    new Clipboard(`.${COPY_TO_CLIPBOARD_CLASS}`);
```
Comment (bartaz) [inline src/common/components/share/index.js]: Will it work just fine if there are two instances of that on a page?

So I suppose `new Clipboard` would be called 2 times with same class name?

Comment (earnubs) [inline src/common/components/share/index.js]: Good point, I was looking at the spec and expecting only one of these per page.

**Comment by earnubs** [comment]
image

File: src/common/components/share/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      this.initClipboard();
+    }
+  }
+
+  componentWillUnmount() {
+    this.clipboard && this.clipboard.destroy();
+  }
+
+  initClipboard() {
+    this.clipboard = new Clipboard(this.copyBtn);
```
Comment (earnubs) [inline src/common/components/share/index.js]: Only add the eventListener to a single component instance, so multiple instances of this component can live with each other.

**Comment by earnubs** [comment]
@bartaz have another look, addressing your comment by using a ref to the component as the target of the clipboard lib rather than a selector, and cleaning up on unmount.

File: src/common/components/share/index.js (hunk 2 above)
Comment (bartaz) [inline src/common/components/share/index.js]: Ok, thanks.

That looks much more encapsulated right now.

**Pull request merged**